compiler/rustc_type_ir/src/structural_impls.rs

```diff
@@ -141,6 +141,14 @@ impl<I: Interner, T: TypeVisitable<I>> TypeVisitable<I> for Vec<T> {
     }
 }
 
+// This trivial implementation that returns the given slice unchanged is the
+// only one that makes sense, because we can't return a new slice.
+impl<I: Interner, T: TypeFoldable<I>> TypeFoldable<I> for &[T] {
```

Comment:
isn't that dangerous because we just ignore `T` here? I would have expected the impl to require `T: !TypeFoldable<I>` to check that it is trivial.

Reply:
Yeah, what is this impl used for? This is certainly not the right implementation (but arguably there should not be one).

Reply:
The impls that the third commit removes: `&'tcx [InlineAsmTemplatePiece]` and `&'tcx [Span]`. These are things carried over from HIR, and treated as constants.

The negative bounds suggestion is a good one, except that they're not supported?
```
error: negative bounds are not supported
   --> compiler/rustc_type_ir/src/structural_impls.rs:146:20
    |
146 | impl<I: Interner, T: !TypeFoldable<I>> TypeFoldable<I> for &[T] {
    |                    ^^^^^^^^^^^^^^^^^^ negative bounds are not supported
```
At this point I think I'll go back to the original commit which just added a comment explaining why `TypeFoldable` isn't impl'd for `&[T]`.